src/main/java/io/xdb/core/client/Client.java

```diff
@@ -0,0 +1,46 @@
+package io.xdb.core.client;
+
+import io.xdb.core.process.BasicClientProcessOptions;
+import io.xdb.core.process.ProcessOptions;
+import io.xdb.core.registry.Registry;
+import io.xdb.core.state.AsyncState;
+import io.xdb.gen.models.AsyncStateConfig;
+import java.util.Optional;
+
+public class Client {
+
+    private final Registry registry;
+    private final ClientOptions clientOptions;
+
+    private final BasicClient basicClient;
+
+    public Client(final Registry registry, final ClientOptions clientOptions) {
+        this.registry = registry;
+        this.clientOptions = clientOptions;
+        this.basicClient = new BasicClient(clientOptions);
+    }
+
+    public String startProcess(
+        final String processType,
```

Comment:
It would be better to use `Process processDefinition` to make it more "strongly typing experience".

Reply:
Sorry 😢 I forgot this. I Java it’s even better to just use the class type like this https://github.com/indeedeng/iwf-java-sdk/blob/d2ae8439d4b7ed996c20303be57f248eb00fe4ef/src/main/java/io/iworkflow/core/Client.java#L63 override process/workflow type is really rare and they can probably use basic client instead

Reply:
Or let's remove the customed type for now. It's really not useful so far. In our new python SDK, I just don't expose the workflowType customization: https://github.com/indeedeng/iwf-python-sdk/blob/main/iwf/workflow.py#L8

It's not late to add it later when needed. 

On the other side, using Class<Process> is much more friendly than using the workflow instance, it save the user code for dependency injection. As a result, I deleted this one in Java SDK recently: https://github.com/indeedeng/iwf-java-sdk/pull/195